src/container.cpp

```diff
@@ -42,34 +41,33 @@ Container::~Container()
 	if (getID() == ITEM_BROWSEFIELD) {
 		g_game.browseFields.erase(getTile());
 
-		for (Item* item : itemlist) {
+		for (const auto& item : itemList) {
 			item->setParent(parent);
 		}
 	} else {
-		for (Item* item : itemlist) {
+		for (const auto& item : itemList) {
 			item->setParent(nullptr);
-			item->decrementReferenceCounter();
 		}
 	}
 }
 
-Item* Container::clone() const
+std::shared_ptr<Item> Container::clone() const
 {
-	Container* clone = static_cast<Container*>(Item::clone());
+	auto clone = std::static_pointer_cast<Container>(Item::clone());
 	if (!clone) {
 		return nullptr;
 	}
 
-	for (Item* item : itemlist) {
+	for (const auto& item : itemList) {
 		clone->addItem(item->clone());
 	}
 	clone->totalWeight = totalWeight;
 	return clone;
 }
 
-Container* Container::getParentContainer()
+std::shared_ptr<Container> Container::getParentContainer()
 {
-	Thing* thing = getParent();
+	auto thing = getParent();
```

Comment:
Could this be `const auto`
```suggestion
	const auto thing = getParent();
```